File: source/BrandonWilliamsCs.EuchreNight/BrandonWilliamsCs.EuchreNight.WebApi/ScoreReports/ScoreReportDto.cs
using BrandonWilliamsCs.EuchreNight.Domain;

namespace BrandonWilliamsCs.EuchreNight.WebApi.ScoreReports;

public class ScoreReportDto
{
  public required Guid UniqueId { get; set; }
  public required Guid SessionId { get; set; }
  public required IList<GameOutcomeDto> Games { get; set; }

  public class GameOutcomeDto
  {
    public required int RoundNumber { get; set; }
    public required int TableNumber { get; set; }
    public required GameOutcome GameOutcome { get; set; }
  }
}

File: source/BrandonWilliamsCS.EuchreNight.WebApp/src/api/ScoreReportDto.ts
export interface ScoreReportDto {
  uniqueId: string;
  sessionId: string;
  games: ScoreReportGameDto[];
}

export interface ScoreReportGameDto {
  roundNumber: number;
  tableNumber: number;
  gameOutcome: GameOutcomeDto;
}

export interface GameOutcomeDto {
  hands: HandOutcomeDto;
  playerScores: Record<number, GamePlayerScoreDto>;
}

export interface GamePlayerScoreDto {
  win: boolean;
  handsWon: number;
  points: number;
  euchres: number;
  lonersAttempted: number;
  lonersWon: number;
}

export interface HandOutcomeDto {
  playerScores: Record<number, HandPlayerScoreDto>;
}

export interface HandPlayerScoreDto {
  win: boolean;
  points: number;
  euchres: number;
  lonersAttempted: number;
  lonersWon: number;
}
